selftests/unit/test_loader.py

```diff
@@ -50,6 +50,45 @@ def test(self):
     main()
 """
 
+AVOCADO_TEST_TAGS = """#!/usr/bin/env python
+from avocado import Test
+from avocado import main
+
+import time
+
+class DisabledTest(Test):
```

Comment:
We've been through this discussion before, and it's a matter of opinion and style.  We should pick and stick to one style.  Look at the contents of `selftests/.data`, and the answer will be pretty clear to our *current* style:

```
-rw-rw-r--. 1 cleber cleber      0 Oct 25 20:43 empty_file
-rw-rw-r--. 1 cleber cleber 378880 Oct 25 20:43 sample.iso
-rw-rw-r--. 1 cleber cleber   1313 Oct  6 10:45 test_archive__symlinks.zip
```

The current style is to keep them as snippets, so that tests are more self contained and even eaiser to follow.  You advocate that "ugly code" goes to snippets, and "nice code" goes to example, then a test that uses both good and bad code is spread all around?

Also, the *current style* is not like that.  See the example from the same file in discussion (`test_loader.py`):

```python
AVOCADO_TEST_OK = """#!/usr/bin/env python
from avocado import Test
from avocado import main

class PassTest(Test):
    def test(self):
        pass

if __name__ == "__main__":
    main()
"""

AVOCADO_TEST_OK_DISABLED = """#!/usr/bin/env python
from avocado import Test
from avocado import main

class PassTest(Test):
    '''
    Instrumented test, but disabled using an Avocado docstring tag
    :avocado: disable
    '''
    def test(self):
        pass

if __name__ == "__main__":
    main()
"""
```

What's ugly about that?

With regards to debugging, I understand you'd rather edit the files and rerun the tests, but then I prefer to edit the snippet and step through it on a debugger.  Conflicting preferences and I don't think mine should take precedence... but the fact is that for other reasons other my personal preference, we decided to keep the snippets approach.